server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/KeyspacesConfiguration.java

```diff
@@ -0,0 +1,148 @@
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.mailbox;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.james.backends.cassandra.init.configuration.KeyspaceConfiguration;
+
+public class KeyspacesConfiguration {
+    public static class Builder {
+        private Optional<String> keyspace;
+        private Optional<String> cacheKeyspace;
+        private Optional<Integer> replicationFactor;
+        private Optional<Boolean> durableWrites;
+
+        public Builder() {
+            keyspace = Optional.empty();
+            cacheKeyspace = Optional.empty();
+            replicationFactor = Optional.empty();
+            durableWrites = Optional.empty();
+        }
+
+        public Builder keyspace(Optional<String> keyspace) {
+            this.keyspace = keyspace;
+            return this;
+        }
+
+        public Builder cacheKeyspace(Optional<String> cacheKeyspace) {
+            this.cacheKeyspace = cacheKeyspace;
+            return this;
+        }
+
+        public Builder cacheKeyspace(String cacheKeyspace) {
+            return cacheKeyspace(Optional.of(cacheKeyspace));
+        }
+
+        public Builder keyspace(String keyspace) {
+            return keyspace(Optional.of(keyspace));
+        }
+
+        public Builder replicationFactor(Optional<Integer> replicationFactor) {
+            this.replicationFactor = replicationFactor;
+            return this;
+        }
+
+        public Builder replicationFactor(int replicationFactor) {
+            return replicationFactor(Optional.of(replicationFactor));
+        }
+
+        public Builder disableDurableWrites() {
+            this.durableWrites = Optional.of(false);
+
+            return this;
+        }
+
+        public KeyspacesConfiguration build() {
+            return new KeyspacesConfiguration(
+                keyspace.orElse(DEFAULT_KEYSPACE),
+                cacheKeyspace.orElse(DEFAULT_CACHE_KEYSPACE),
```

Comment:
Should we ensure that cacheKeySpace as a different name than the main keyspace ?